gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/shift/CountryBalanceComputation.java

```diff
@@ -28,25 +31,53 @@
 /**
  * @author Ameni Walha {@literal <ameni.walha at rte-france.com>}
  */
+@Service
 public final class CountryBalanceComputation {
     private static final Logger LOGGER = LoggerFactory.getLogger(CountryBalanceComputation.class);
 
-    private CountryBalanceComputation() {
-         // Should not be instantiated
+    private final Logger businessLogger;
+
+    public CountryBalanceComputation(Logger businessLogger) {
+        this.businessLogger = businessLogger;
     }
 
-    public static Map<String, Double> computeSweCountriesBalances(Network network) {
+    public Map<String, Double> computeSweCountriesBalances(Network network) {
         LOGGER.info("Computing initial SWE countries balance");
         Map<String, Double> countriesBalances = new HashMap<>();
-        runLoadFlow(network, network.getVariantManager().getWorkingVariantId());
-        Map<String, Double> bordersExchanges = computeSweBordersExchanges(network);
-        countriesBalances.put(toEic("PT"),  -bordersExchanges.get("ES_PT"));
-        countriesBalances.put(toEic("ES"), bordersExchanges.values().stream().reduce(0., Double::sum));
-        countriesBalances.put(toEic("FR"), -bordersExchanges.get("ES_FR"));
+
+        String portugalEic = toEic("PT");
+        String spainEic = toEic("ES");
+        String franceEic = toEic("FR");
+
+        if (runLoadFlow(network, network.getVariantManager().getWorkingVariantId())) {
+            Map<String, Double> bordersExchanges = computeSweBordersExchanges(network);
+            countriesBalances.put(portugalEic, -bordersExchanges.get("ES_PT"));
+            countriesBalances.put(spainEic, bordersExchanges.values().stream().reduce(0., Double::sum));
+            countriesBalances.put(franceEic, -bordersExchanges.get("ES_FR"));
+            businessLogger.info("Base case loadflow is secure");
+            return countriesBalances;
+        }
+        LOGGER.warn("Base case is not secure: using net interchange values (Loadflow computation diverged on network '{}')", network.getId());
```

Comment:
Maybe it is more readable with a "else" instead of "guessing" that if the runLoadFlow method returns false, we go to those lines